graphqlapi/src/test/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/BulkMutationFetcherTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.cqlfirst.dml.fetchers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import graphql.ExecutionResult;
+import graphql.GraphQLError;
+import io.stargate.db.schema.Schema;
+import io.stargate.graphql.schema.SampleKeyspaces;
+import io.stargate.graphql.schema.cqlfirst.dml.DmlTestBase;
+import java.util.Collections;
+import org.apache.cassandra.stargate.db.ConsistencyLevel;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public class BulkMutationFetcherTest extends DmlTestBase {
+  @Override
+  public Schema getCQLSchema() {
+    return Schema.create(Collections.singleton(SampleKeyspaces.LIBRARY));
+  }
+
+  @Test
+  @DisplayName("Atomic bulk mutations with single selection should use normal execution")
+  public void mutationAtomicBulkSingleSelectionTest() {
+    assertQuery(
+        "mutation @atomic { m1: bulkInsertbooks(values: [{ title: \"a\" }] ) { applied } }",
+        "INSERT INTO library.books (title) VALUES ('a')");
+  }
+
+  @Test
+  @DisplayName(
+      "Atomic bulk mutations with single selection with N values should use normal execution")
+  public void mutationAtomicBulkSingleSelectionTestNValues() {
```

Comment:
for all mock tests that are using bulk with N elements in the array, I am getting a weird error, I.e.:
`Incompatible conditions title = a and title = b`
from the `WhereProcessor`:
```
format("Incompatible conditions %s and %s", existing.firstCondition, condition));
```
It seems that the insert here treated as one query with multiple AND statements. 
What Am I missing? Inserting N elements in array works in the integration test (see for example: `bulkInsertProductsWithAtomic`) Is the below syntax correct?
```
 "mutation @atomic { m1: bulkInsertbooks(values: [{ title: \"a\" }, { title: \"b\" }] ) { applied } }"
```
@olim7t

Reply:
I can't reproduce it, I'm getting another error because `batchCaptor` didn't capture anything... I suppose that was a different version of the code. But anyway I would try debugging to see how the queries are being constructed.

Reply:
Yes, I am getting the same error from `captor`. However, the mentioned `Incompatible conditions` is also generated but it is not propagated to the caller.

Reply:
There was some problem with mocks for that test. I've moved both test scenarios to the Integration test suit. Removing unit for the same use-case